compose/utils.py

```diff
@@ -60,13 +64,21 @@ def split_buffer(stream, splitter=None, decoder=lambda a: a):
             yield item
 
     if buffered:
-        yield decoder(buffered)
+        try:
+            yield decoder(buffered)
+        except ValueError as e:
+            log.error(
+                'Compose tried parsing the following chunk as a JSON object, '
+                'but failed:\n%s' % repr(buffered)
+            )
+            raise StreamParseError(e)
```

Comment:
I don't think this is correct. It assumes `decoder()` is always going to return a `ValueError`, but it's going to depend on what function is passed as the decoder.

Is this still necessary with the fix below?